src/vs/workbench/services/positronConsole/browser/positronConsoleService.ts

```diff
@@ -2081,36 +2084,48 @@ class PositronConsoleInstance extends Disposable implements IPositronConsoleInst
 	/**
 	 * Executes code.
 	 * @param code The code to execute.
+	 * @param runtimeCodeExecutionMode Possible code execution modes for a language runtime
+	 * runtimeCodeExecutionMode
 	 */
-	private doExecuteCode(code: string) {
+	private doExecuteCode(code: string, runtimeCodeExecutionMode?: RuntimeCodeExecutionMode) {
```

Comment:
This is where the meat of the changes are that need eyes.

It seems like the `mainThreadLanguageRuntime` is allowed to call the session directly but extensions aren't and have to go through the `mainThreadLanguageRuntime` which ends up going through this `positronConsoleService` which is why we need to handle the different `RuntimeCodeExecutionMode` consequences here.

My understanding is that the `_runtimeItemActivities` list contains the information rendered by the console UI, aka the code that will be executed and its output. The list is modified in the service here and we need to conditionally decide when to add things to the list based off the `RuntimeCodeExecutionMode` value.

The console history list is created on the component side and relies on the `executionHistoryService` to create the history initially and the `onDidExecuteCode` event to update the list. The only listeners I found for this event was in the console so it should be fine if we don't fire it when we don't want to add input to the history. 

I haven't fully dug into how the `executionHistoryService` creates the initial history list now that I am reflecting back on these changes so I don't know if I've handled all the scenarios for the console input history case. My testing didn't reveal anything but I'll take another look into this to make sure I didn't miss anything